openstack/container/experimental/capsules/results.go

```diff
@@ -0,0 +1,106 @@
+package capsules
+
+import (
+	"encoding/json"
+	"time"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+type commonResult struct {
+	gophercloud.Result
+}
+
+// Extract is a function that accepts a result and extracts a capsule resource.
+func (r commonResult) Extract() (*Capsule, error) {
+	var s *Capsule
+	err := r.ExtractInto(&s)
+	return s, err
+}
+
+// GetResult represents the result of a get operation.
+type GetResult struct {
+	commonResult
+}
+
+// Represents a Container Orchestration Engine Bay, i.e. a cluster
+type Capsule struct {
+	// UUID for the capsule
+	UUID string `json:"uuid"`
+
+	// ID for the capsule
+	ID int `json:"id"`
+
+	// User ID for the capsule
+	UserID string `json:"user_id"`
+
+	// Project ID for the capsule
+	ProjectID string `json:"project_id"`
+
+	// cpu for the capsule
+	CPU float64 `json:"cpu"`
+
+	// Memory for the capsule
+	Memory string `json:"memory"`
+
+	// The name of the capsule
+	MetaName string `json:"meta_name"`
+
+	// Indicates whether capsule is currently operational. Possible values include:
+	// Running,
+	Status string `json:"status"`
+
+	// The created time of the capsule.
+	CreatedAt time.Time `json:"-"`
+
+	// The updated time of the capsule.
+	UpdatedAt time.Time `json:"-"`
+
+	// Links includes HTTP references to the itself, useful for passing along to
+	// other APIs that might want a server reference.
+	Links []interface{} `json:"links"`
+
+	// The capsule version
+	CapsuleVersion string `json:"capsule_version"`
+
+	// The capsule restart policy
+	RestartPolicy string `json:"restart_policy"`
+
+	// The capsule metadata labels
+	MetaLabels map[string]string `json:"meta_labels"`
+
+	// The list of containers uuids inside capsule.
+	ContainersUUIDs []string `json:"containers_uuids"`
+
+	// The capsule IP addresses
+	Addresses map[string][]Address `json:"addresses"`
+
+	// The capsule volume attached information
+	VolumesInfo map[string][]string `json:"volumes_info"`
+}
+
+type Address struct {
```

Comment:
I'm seeing a `subnet_id` field being returned:

```json
    "addresses": {                                                  
        "1883661f-ddb9-4414-81b6-e4e0b880a14f": [                   
            {                                                       
                "addr": "172.24.4.9",                               
                "port": "ba0accd0-f3cd-49af-b745-42b650704344",     
                "preserve_on_delete": false,                        
                "subnet_id": "e26b1199-4a6d-4279-a122-feb72691d8ae",
                "version": 4                                        
            }                                                       
        ]                                                           
    },                                                              
```

Reply:
Yes, the subnet_id field is recently merged. I will add.

Reply:
For addresses, there is a new [patch](https://review.openstack.org/542604) added subnet yesterday. I will add corresponding support.

Reply:
hah - how coincidental 😄 

sounds good.